src/bp-members/bp-members-filters.php

```diff
@@ -830,39 +830,56 @@ function bb_repair_member_profile_links_callback( $is_background = false, $paged
 		return;
 	}
 
+	global $wpdb;
+
+	// Define default values.
 	$per_page = 50;
+	$offset   = 0;
 
-	if ( $is_background ) {
-		$offset = ( ( $paged - 1 ) * $per_page );
-	} else {
-		// phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
-		$offset = isset( $_POST['offset'] ) ? (int) ( $_POST['offset'] ) : 0;
-	}
+	if ( ! $is_background ) {
 
-	$args = array(
-		'fields' => 'ID',
-		'number' => $per_page,
-		'offset' => $offset,
-	);
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		$offset   = isset( $_POST['offset'] ) ? (int) ( $_POST['offset'] ) : 0;
```

Comment:
Where from offset is coming? isn't we suppose to use $paged parameter?

Reply:
Please describe why the $_POST offset is being used. maybe I am not aware.

Reply:
The offset comes from the tools where we pass in the results after the process is done.
Here is the code: https://github.com/buddyboss/buddyboss-platform/blob/5de9658aad2803b0edb06b8f7937d034fe9c251d/src/bp-members/bp-members-filters.php#L886-L897